### ISSUE TITLE
kola: rawhide: 20220118 aarch64 multipathd service failing

### ISSUE THREAD

**Issue by dustymabe** [coreos/fedora-coreos-tracker#1071]
Our rawhide aarch64 builds [started failing](https://jenkins-fedora-coreos.apps.ocp.ci.centos.org/blue/organizations/jenkins/multi-arch-pipeline/detail/multi-arch-pipeline/411) because `multipathd` won't start:

```
[core@cosa-devsh ~]$ systemctl status multipathd.service | cat
× multipathd.service - Device-Mapper Multipath Device Controller
     Loaded: loaded (/usr/lib/systemd/system/multipathd.service; enabled; vendor preset: enabled)
     Active: failed (Result: signal) since Wed 2022-01-19 02:55:16 UTC; 1min 14s ago
TriggeredBy: ○ multipathd.socket
  Condition: start condition failed at Wed 2022-01-19 02:55:44 UTC; 46s ago
             └─ ConditionPathExists=/etc/multipath.conf was not met
   Main PID: 553 (code=killed, signal=SEGV)
        CPU: 67ms

Jan 19 02:55:16 localhost systemd[1]: Starting multipathd.service - Device-Mapper Multipath Device Controller...
Jan 19 02:55:16 localhost systemd[1]: Started multipathd.service - Device-Mapper Multipath Device Controller.
Jan 19 02:55:16 localhost systemd[1]: multipathd.service: Main process exited, code=killed, status=11/SEGV
Jan 19 02:55:16 localhost systemd[1]: multipathd.service: Failed with result 'signal'.
Jan 19 02:55:44 localhost systemd[1]: multipathd.service - Device-Mapper Multipath Device Controller was skipped because of a failed condition check (ConditionPathExists=/etc/multipath.conf).
```

Full logs: [kola.zip](https://github.com/coreos/fedora-coreos-tracker/files/7893742/kola.zip)

I was able to reproduce in a local test. The package set in this update looks like:

```
[2022-01-18T23:29:13.610Z] Upgraded:
[2022-01-18T23:29:13.610Z]   NetworkManager 1:1.36.0-0.3.fc36 -> 1:1.36.0-0.4.fc36
[2022-01-18T23:29:13.611Z]   NetworkManager-cloud-setup 1:1.36.0-0.3.fc36 -> 1:1.36.0-0.4.fc36
[2022-01-18T23:29:13.611Z]   NetworkManager-libnm 1:1.36.0-0.3.fc36 -> 1:1.36.0-0.4.fc36
[2022-01-18T23:29:13.611Z]   NetworkManager-team 1:1.36.0-0.3.fc36 -> 1:1.36.0-0.4.fc36
[2022-01-18T23:29:13.611Z]   NetworkManager-tui 1:1.36.0-0.3.fc36 -> 1:1.36.0-0.4.fc36
[2022-01-18T23:29:13.611Z]   afterburn 5.1.0-3.fc36 -> 5.2.0-1.fc36
[2022-01-18T23:29:13.611Z]   afterburn-dracut 5.1.0-3.fc36 -> 5.2.0-1.fc36
[2022-01-18T23:29:13.611Z]   bash 5.1.8-3.fc36 -> 5.1.16-1.fc36
[2022-01-18T23:29:13.611Z]   btrfs-progs 5.15.1-1.fc36 -> 5.16-1.fc36
[2022-01-18T23:29:13.611Z]   crun 1.4-1.fc36 -> 1.4.1-1.fc36
[2022-01-18T23:29:13.611Z]   cryptsetup 2.4.2-1.fc36 -> 2.4.3-1.fc36
[2022-01-18T23:29:13.611Z]   cryptsetup-libs 2.4.2-1.fc36 -> 2.4.3-1.fc36
[2022-01-18T23:29:13.611Z]   cups-libs 1:2.4.0-1.fc36 -> 1:2.4.0-2.fc36
[2022-01-18T23:29:13.611Z]   dnsmasq 2.86-3.fc36 -> 2.86-4.fc36
[2022-01-18T23:29:13.611Z]   efivar-libs 37-17.fc35 -> 38-1.fc36
[2022-01-18T23:29:13.611Z]   expat 2.4.1-2.fc35 -> 2.4.3-1.fc36
[2022-01-18T23:29:13.611Z]   flatpak-session-helper 1.12.2-1.fc36 -> 1.12.3-1.fc36
[2022-01-18T23:29:13.611Z]   fwupd 1.7.3-1.fc36 -> 1.7.4-1.fc36
[2022-01-18T23:29:13.611Z]   git-core 2.34.1-1.fc36 -> 2.35.0-0.1.rc1.fc36
[2022-01-18T23:29:13.611Z]   grub2-common 1:2.06-12.fc36 -> 1:2.06-13.fc36
[2022-01-18T23:29:13.611Z]   grub2-efi-aa64 1:2.06-12.fc36 -> 1:2.06-13.fc36
[2022-01-18T23:29:13.611Z]   grub2-tools 1:2.06-12.fc36 -> 1:2.06-13.fc36
[2022-01-18T23:29:13.611Z]   grub2-tools-minimal 1:2.06-12.fc36 -> 1:2.06-13.fc36
[2022-01-18T23:29:13.611Z]   kernel 5.16.0-60.fc36 -> 5.17.0-0.rc0.20220112gitdaadb3bd0e8d.63.fc36
[2022-01-18T23:29:13.611Z]   kernel-core 5.16.0-60.fc36 -> 5.17.0-0.rc0.20220112gitdaadb3bd0e8d.63.fc36
[2022-01-18T23:29:13.611Z]   kernel-modules 5.16.0-60.fc36 -> 5.17.0-0.rc0.20220112gitdaadb3bd0e8d.63.fc36
[2022-01-18T23:29:13.611Z]   kexec-tools 2.0.23-1.fc36 -> 2.0.23-2.fc36
[2022-01-18T23:29:13.611Z]   libfido2 1.9.0-1.fc36 -> 1.10.0-1.fc36
[2022-01-18T23:29:13.611Z]   libgcc 11.2.1-7.fc36 -> 12.0.0-0.5.fc36
[2022-01-18T23:29:13.611Z]   libgomp 11.2.1-7.fc36 -> 12.0.0-0.5.fc36
[2022-01-18T23:29:13.611Z]   libibverbs 38.1-1.fc36 -> 38.1-2.fc36
[2022-01-18T23:29:13.611Z]   libstdc++ 11.2.1-7.fc36 -> 12.0.0-0.5.fc36
[2022-01-18T23:29:13.611Z]   libxcrypt 4.4.27-1.fc36 -> 4.4.27-2.fc36
[2022-01-18T23:29:13.611Z]   moby-engine 20.10.11-1.fc36 -> 20.10.12-1.fc36
[2022-01-18T23:29:13.611Z]   readline 8.1-3.fc35 -> 8.1-4.fc36
[2022-01-18T23:29:13.611Z]   rpm 4.17.0-1.fc36.1 -> 4.17.0-4.fc36
[2022-01-18T23:29:13.611Z]   rpm-libs 4.17.0-1.fc36.1 -> 4.17.0-4.fc36
[2022-01-18T23:29:13.611Z]   rpm-plugin-selinux 4.17.0-1.fc36.1 -> 4.17.0-4.fc36
[2022-01-18T23:29:13.611Z]   runc 2:1.1.0-0.1.rc.1.fc36 -> 2:1.1.0-1.fc36
[2022-01-18T23:29:13.611Z]   selinux-policy 35.8-1.fc36 -> 35.9-1.fc36
[2022-01-18T23:29:13.611Z]   selinux-policy-targeted 35.8-1.fc36 -> 35.9-1.fc36
[2022-01-18T23:29:13.611Z]   shadow-utils 2:4.9-8.fc36 -> 2:4.9-9.fc36
[2022-01-18T23:29:13.611Z]   shadow-utils-subid 2:4.9-8.fc36 -> 2:4.9-9.fc36
[2022-01-18T23:29:13.611Z]   ssh-key-dir 0.1.2-8.fc35 -> 0.1.2-9.fc36
[2022-01-18T23:29:13.611Z]   vim-data 2:8.2.4051-1.fc36 -> 2:8.2.4068-1.fc36
[2022-01-18T23:29:13.611Z]   vim-minimal 2:8.2.4051-1.fc36 -> 2:8.2.4068-1.fc36
[2022-01-18T23:29:13.611Z] Added:
[2022-01-18T23:29:13.611Z]   nano-6.0-1.fc36.aarch64
[2022-01-18T23:29:13.611Z]   nano-default-editor-6.0-1.fc36.noarch
```

**Comment by dustymabe** [comment]
Thanks @ravanelli for looking into this!

**Comment by ravanelli** [comment]
Looks in the past there was a similar bug for multipath https://bugzilla.redhat.com/show_bug.cgi?id=1508483 
Let me try to downgrade some packages here to see how that goes.

**Comment by ravanelli** [comment]
I tried to downgrade some packages, but none of them worked:  Kernel, selinux, afterburn, grub2 and NetworkManager 

```
Journal error: 
[   12.275299] udevadm[426]: systemd-udev-settle.service is deprecated. Please fix multipathd-configure.service, multipathd.service not to pull it in.
```


```
[core@cosa-devsh ~]$ systemctl list-dependencies multipathd.service
multipathd.service
● ├─system.slice
● ├─systemd-udev-settle.service
● └─systemd-udev-trigger.service


root@cosa-devsh core]#  multipath -l
81.831023 | /etc/multipath.conf does not exist, blacklisting all devices.
81.832247 | You can run "/sbin/mpathconf --enable" to create
81.833176 | /etc/multipath.conf. See man mpathconf(8) for more details

[root@cosa-devsh core]# /sbin/mpathconf
multipath is enabled
find_multipaths is yes
user_friendly_names is enabled
default property blacklist is disabled
enable_foreign is not set (no foreign multipath devices will be shown)
dm_multipath module is loaded
multipathd is not running

[root@cosa-devsh core]# /sbin/mpathconf --enable
[root@cosa-devsh core]# multipath -l
```
Running  `/mpathconf --enable` persists across reboot

**Comment by lucab** [comment]
From the kola logs, do note that the segfault is happening in the initramfs, when the service is stopping:
```
Jan 18 23:43:44.154727 multipathd[442]: exit (signal)
Jan 18 23:43:44.154727 multipathd[442]: --------shut down-------
Jan 18 23:43:44.149000 audit[442]: ANOM_ABEND auid=4294967295 uid=0 gid=0 ses=4294967295 subj=kernel pid=442 comm="multipathd" exe="/usr/sbin/multipathd" sig=11 res=1
Jan 18 23:43:44.236235 systemd[1]: Reached target initrd-fs.target - Initrd File Systems.
Jan 18 23:43:44.244684 systemd[1]: Reached target initrd.target - Initrd Default Target.
Jan 18 23:43:44.281778 systemd-coredump[733]: Failed to connect to coredump service: No such file or directory
Jan 18 23:43:44.457878 systemd[1]: multipathd.service: Main process exited, code=killed, status=11/SEGV
Jan 18 23:43:44.463689 systemd[1]: multipathd.service: Failed with result 'signal'.
Jan 18 23:43:44.472406 systemd[1]: Stopped multipathd.service - Device-Mapper Multipath Device Controller.
```

**Comment by ravanelli** [comment]
I was looking for the package deplist for multipath and downgrading `libgcc` to `11.2.1-7.fc36` solved the issue.  I will open a PR for it.

**Comment by dustymabe** [comment]
Looks like @jcajka built the `gcc` that produced `libgcc-12.0.0-0.5.fc36`: https://koji.fedoraproject.org/koji/buildinfo?buildID=1878152. Maybe he can help figure out the root cause.

**Comment by ravanelli** [comment]
BZ opened: https://bugzilla.redhat.com/show_bug.cgi?id=2044446

**Comment by dustymabe** [comment]
This seems to be fixed with `libcc-12.0.1-0.3.fc36`.

**Comment by dustymabe** [comment]
Fixed in https://github.com/coreos/fedora-coreos-config/pull/1459
